adapters/repos/db/lsmkv/segment_group_compaction.go

```diff
@@ -262,6 +262,30 @@ func (sg *SegmentGroup) replaceCompactedSegments(old1, old2 int,
 		sg.segments[old2].countNetAdditions
 	sg.maintenanceLock.RUnlock()
 
+	sg.maintenanceLock.Lock()
```

Comment:
Can this be just a RLock()? as we only read data from the segments slice

Reply:
My goal would be to avoid parallel compactions, which could lead to two processes working with the same `.tmp` files. But maybe we should add a prefix to the tmp segment id in those scenarios.